src/Voron/Impl/FreeSpace/StreamBitArray.cs

```diff
@@ -113,6 +108,196 @@ public bool Get(int index)
             return (_inner[index >> 5] & (1 << (index & 31))) != 0;
         }
 
+        public int? GetContinuousRangeStart(int num)
+        {
+            return num switch
+            {
+                1 => FirstSetBit(),
+                < BitsInWord => FindSmallRange(num),
+                < 64 => FindLargeRange<int>(num),
+                _ => FindLargeRange<long>(num)
+            };
+        }
+
+        private int? FirstSetBit()
+        {
+            if (Avx2.IsSupported)
```

Comment:
This is actively harmful. The JIT would generate the right code here for the platform.
For example, it will generate ARM vector, not using Avx2

Basically, we need _just_ the vector code, there is no point in doing scalar version as well, or using `if`

Reply:
done